ethereum.module

```diff
@@ -45,3 +47,13 @@ function ethereum_field_formatter_info_alter(array &$info) {
   // formatter.
   $info['basic_string']['field_types'][] = 'ethereum_address';
 }
+
+/**
+ * Implements hook_js_settings_alter().
+ */
+function ethereum_js_settings_alter(&$settings, AttachedAssetsInterface $assets) {
+  // Provide the URL of the current server in drupalSettings to allow JavaScript
+  // code to rely on an instantiated Web3 object.
+  $current_server = \Drupal::config('ethereum.settings')->get('current_server');
+  $settings['ethereum']['currentServerHost'] = EthereumServer::load($current_server)->getUrl();
```

Comment:
We risk publishing private Server information. 
If we provide a public url to connect it might be a different from the backend.

Reply:
Good point! I added a new 'frontend_server' setting so we can select different servers for backend / frontend.